File: an2_sem4/MPP/Homework/sapt78/cs/FestivalSellpoint/FestivalSellpoint.Network/DTO/AngajatDTO.cs
using FestivalSellpoint.Domain;
using System;

namespace FestivalSellpoint.Network.DTO
{
    [Serializable]
    public class AngajatDTO : EntityDTO
    {
        public string Username { get; }
        public string Password { get; }

        public AngajatDTO(string username, string password)
        {
            Username = username;
            Password = password;
        }

        public static AngajatDTO FromAngajat(Angajat a)
            => new AngajatDTO(a.Username, a.Password) { Id = a.Id };

        public Angajat ToAngajat()
            => new Angajat(Username, Password, "dummy@notusedemail.com") { Id = Id };
    }
}

File: an2_sem4/MPP/Homework/sapt10/java/network/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/network/dto/AngajatDTO.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.dto;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Angajat;

public class AngajatDTO extends EntityDTO {
    private String username;
    private String password;

    public AngajatDTO() {}

    public AngajatDTO(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public static AngajatDTO fromAngajat(Angajat angajat){
        var a = new AngajatDTO(angajat.getUsername(), angajat.getPassword());
        a.setId(angajat.getId());
        return a;
    }

    public Angajat toAngajat(){
        var a = new Angajat(getUsername(), getPassword(), "dummy@notusedemail.com");
        a.setId(getId());
        return a;
    }

    @Override
    public String toString() {
        return "AngajatDTO{" +
                "username='" + username + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}
